src/comm/LinkManager.cc

```diff
@@ -426,6 +426,9 @@ void LinkManager::loadLinkConfigurationList()
                                 pLink->setAutoConnect(autoConnect);
                                 pLink->loadSettings(settings, root);
                                 addConfiguration(pLink);
+                                if (pLink->isAutoConnect()) {
+                                    createConnectedLink(pLink);
```

Comment:
The loadLinkConfigurationList() function is called (from QGCApplication.cc) before checking to see if the settings are to be cleared, which would invalidate the settings loaded. The code above seems fine but the function call in QGCApplication.cc has to be moved after testing for `_settingsUpgraded`.